### PR TITLE
Add affinity to gateway proxy helm values

### PR DESCRIPTION
**Pull request by kevin-shelaga**
# Description

Added new Affinity helm value on gatewayproxy, to facilitate more fine grained Affinity rules

# Context

PR for https://github.com/solo-io/gloo/issues/3995

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3995

### CODE REVIEW

File: install/helm/gloo/templates/7-gateway-proxy-deployment.yaml [hunk 1]
```diff
@@ -102,6 +102,9 @@ spec:
                   gloo: gateway-proxy
               topologyKey: kubernetes.io/hostname
 {{- end}}
+{{- if $spec.extraAntiAffinity }}
```
Comment (kdorosh) [inline install/helm/gloo/templates/7-gateway-proxy-deployment.yaml:105]: Perhaps we can cal this affinity instead of anti affinity?

**Review by kdorosh — CHANGES_REQUESTED** [review]
Naming not, name of the helm values and then its use in test is a bit confusing

File: install/helm/gloo/templates/7-gateway-proxy-deployment.yaml (hunk 1 above)
Comment (kevin-shelaga) [reply install/helm/gloo/templates/7-gateway-proxy-deployment.yaml]: changed to affinity, much better

**Review by kdorosh — APPROVED** [review]

**Pull request merged**